File: sp48/crypt/prohorow/test4.c
/****************************************************************************/
/*                                                                          */
/*      Data Encryption ToolKit. Version 1.3                                */
/*                                                                          */
/*      test4.c -- test module for DES module 4                             */
/*                                                                          */
/*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    */
/*                                                                          */
/*      Purpose:                                                            */
/*        Test encrypt_program and decrypt_program functions. File          */
/*        test4.exe must be encrypted from test_value_1 to key and          */
/*        from test_value_2 to idle by "Prokhorow" key.                     */
/*                                                                          */
/****************************************************************************/


#include <stdio.h>
#include <dos.h>
#include "encrypt.h"

int test_value_1[5] = {1, 2, 3, 4, 5};
char key[] = "Prokhorow";
int test_value_2[5] = {6, 7, 8, 9, 10};
char idle = 0;
const int KEY_NUMBER = 1;
char key_area[ENCRYPTION_KEY_AREA_SIZE (KEY_NUMBER)];

ENCRYPTION_CONTROL_AREA (0, 3)

void test_write () {
  for (int i = 0; i < 5; i ++)
    printf ("%i ", test_value_1[i]);
  for (i = 0; i < 5; i ++)
    printf ("%i ", test_value_2[i]);
  printf ("\n");
  }

void main () {
  test_write ();
  decryption_key (key_area, key);
  decrypt_program (key_area, self_encrypt_start, encryption_control_area_0);
  test_write ();
  encryption_key (key_area, key);
  encrypt_program (key_area, self_encrypt_start, encryption_control_area_0);
  test_write ();
  }

File: sp48/crypt/prohorow/test4.pas
{****************************************************************************}
{*                                                                          *}
{*      Data Encryption ToolKit. Version 1.3                                *}
{*                                                                          *}
{*      test5.pas -- test module for DES module 5                           *}
{*                                                                          *}
{*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    *}
{*                                                                          *}
{*      Purpose:                                                            *}
{*        Test EncryptProgram and DecryptProgram procedures. File           *}
{*        test4.exe must be encrypted from TestValue1 to Key and from       *}
{*        TestValue2 to EncryptionControlArea0 by 'Prokhorow' key.          *}
{*                                                                          *}
{****************************************************************************}


program Test4;

  uses Encrypt;

  const
    TestValue1: array [1..5] of Integer = (1, 2, 3, 4, 5);
    Key: String = 'Prokhorow';
    TestValue2: array [1..5] of Integer = (6, 7, 8, 9, 10);
    EncryptionKeyNumber = 1;
    EncryptionFragmentNumber0 = 3;

{$i encrcon.inc}
{$i encrca0.inc}

  var
    KeyArea: array [1..EncryptionKeyAreaSize] of Char;

  procedure TestWrite;

    var
      I: Integer;

    begin
      for I := 1 to 5 do
        Write (TestValue1[I], ' ');
      for I := 1 to 5 do
        Write (TestValue2[I], ' ');
      WriteLn
    end;

  begin
    TestWrite;
    DecryptionKey (KeyArea, Key);
    DecryptProgram (KeyArea, SelfEncryptStart, EncryptionControlArea0);
    TestWrite;
    EncryptionKey (KeyArea, Key);
    EncryptProgram (KeyArea, SelfEncryptStart, EncryptionControlArea0);
    TestWrite
  end.
